lib/core/src/server/build-static.js

```diff
@@ -72,42 +40,105 @@ export async function buildStaticStandalone(options) {
+      managerConfig =>
+        new Promise((resolve, reject) => {
+          webpack(managerConfig).run((err, stats) => {
+            if (err || !stats || stats.hasErrors()) {
+              logger.error('Failed to build the manager');
+              // eslint-disable-next-line no-unused-expressions
+              err && logger.error(err.message);
+              // eslint-disable-next-line no-unused-expressions
+              stats && stats.hasErrors() && stats.toJson().errors.forEach(e => logger.error(e));
+              process.exitCode = 1;
+
+              reject(stats);
+            }
+
+            resolve(stats);
+          });
+        })
+    )
+    .then(stats => {
+      logger.trace({ message: 'manager built', time: process.hrtime(startTime) });
+      logger.info(stats.hash);
+      stats.toJson().warnings.forEach(e => logger.warn(e));
+    })
+    .then(() => {
+      startTime = process.hrtime();
+      logger.info('building preview..');
+    })
+    .then(() =>
+      loadConfig({
+        configType,
+        outputDir,
+        packageJson,
+        corePresets: [require.resolve('./preview/preview-preset.js')],
+        overridePresets: [require.resolve('./preview/custom-webpack-preset.js')],
+        ...options,
+      })
+    )
+    .then(
+      config =>
+        new Promise((resolve, reject) => {
+          webpack(config).run((err, stats) => {
+            if (err || !stats || stats.hasErrors()) {
+              logger.error('Failed to build the preview');
+              // eslint-disable-next-line no-unused-expressions
+              err && logger.error(err.message);
+              // eslint-disable-next-line no-unused-expressions
+              stats && stats.hasErrors() && stats.toJson().errors.forEach(e => logger.error(e));
+              process.exitCode = 1;
+              reject(err);
+            }
+
+            resolve(stats);
+          });
+        })
+    )
+    .then(stats => {
+      logger.trace({ message: 'preview built', time: process.hrtime(startTime) });
+      logger.info(stats.hash);
+      stats.toJson().warnings.forEach(e => logger.warn(e));
+    })
+    .then(() => {
+      logger.info(`output directory: ${outputDir}`);
+    })
+    .catch(p => {
+      if (p && p.toJson) {
+        const stats = p.toJson();
+
+        stats.errors.forEach(e => logger.error(e));
+        stats.warnings.forEach(e => logger.error(e));
+      } else {
+        logger.error(p);
       }
-
-      const previewTotalTime = process.hrtime(previewStartTime);
-      logger.trace({ message: 'preview built', time: previewTotalTime });
-
-      return resolve(stats);
-    };
-
-    logger.info('building preview..');
-    const compiler = webpack(config);
-
-    if (watch) {
-      compiler.watch({}, webpackCb);
-    } else {
-      compiler.run(webpackCb);
-    }
-  });
+    });
 }
 
-export async function buildStatic({ packageJson, ...loadOptions }) {
+export function buildStatic({ packageJson, ...loadOptions }) {
```

Comment:
since this function returns a promise I think it should be async still

Reply:
`async` should only be used in combination with `await` right?

Reply:
The typescript compiler is becoming part of my brain! https://javascript.info/async-await The `async` keyword on a function just means that the function will always return a promise. It doesn't have to be used in combination with `await` at all. 

Probably an unneeded change, but it does communicate easily that this function returns a promise